x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/BaseElasticsearchInternalService.java

```diff
@@ -221,26 +229,29 @@ public static String selectDefaultModelVariantBasedOnClusterArchitecture(
         }
     }
 
-    private void platformArchitecture(ActionListener<Set<String>> platformArchitectureListener) {
+    private void platformArchitecture(ActionListener<PreferredModelVariant> platformArchitectureListener) {
         // Find the cluster platform as the service may need that
         // information when creating the model
         MlPlatformArchitecturesUtil.getMlNodesArchitecturesSet(
             platformArchitectureListener.delegateFailureAndWrap((delegate, architectures) -> {
-                if (architectures.isEmpty() && clusterIsInElasticCloud()) {
-                    // In Elastic cloud ml nodes run on Linux x86
-                    delegate.onResponse(Set.of("linux-x86_64"));
+                if (architectures.isEmpty() && isClusterInElasticCloud()) {
+                    // There are no ml nodes to check the current arch.
+                    // However, in Elastic cloud ml nodes run on Linux x86
+                    delegate.onResponse(PreferredModelVariant.LINUX_X86_OPTIMIZED);
                 } else {
-                    delegate.onResponse(architectures);
+                    delegate.onResponse(PreferredModelVariant.PLATFORM_AGNOSTIC);
                 }
             }),
             client,
             inferenceExecutor
         );
     }
 
-    static boolean clusterIsInElasticCloud() {
-        // use a heuristic to determine if in Elastic cloud.
-        return true; // TODO
+    boolean isClusterInElasticCloud() {
+        // Use the ml lazy node count as a heuristic
+        // to determine if in Elastic cloud.
+        var maxMlLazyNodes = clusterService.getClusterSettings().get(MachineLearningField.MAX_LAZY_ML_NODES);
```

Comment:
I have no clue why this would be a reasonable heuristic. Could you explain that in the comments?

Reply:
If `MAX_LAZY_ML_NODES`  > 0 then it means scaling is enable and some external service can add more ml nodes to the cluster up to the value of `MAX_LAZY_ML_NODES` . I added a comment to that effect

Search for `xpack.ml.max_lazy_ml_nodes`  in https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-settings.html